### ISSUE TITLE
 change in bookmarks not syncing

### ISSUE THREAD

**Issue by usernameandpassword1** [mozilla-mobile/firefox-ios#8706]
## Steps to reproduce
1-with a signed in profile, make a change to add or remove bookmarks(specifically bookmarks toolbar)
2-try to use any other device with the same signed in profile and manually click sync now(with the bookmarks option checked on all devices)

### Expected behavior
After following the 2 step process above, the expected behavior would be for a seem less sync of bookmarks

### Actual behavior
After following 2 step behavior above, the actual behavior is that the syncing does not occur unless the sync now option is initiated from source of change device. Only thereafter the change will be reflected on the other devices whether with the manual or automatic sync 

### Device & build information
* Device: ? iPad Pro 12.9 inch 3rd gen iPhone 11 Pro Max
* Build version: ? 14.6 for both
### Notes
Attachments: 
Simply put the changes for bookmarks won’t be reflected in the users profile database unless the sync now option is clicked from the source of change. This is not ideal since there would be times the user may forget to click the sync now option after a change is made in the bookmarks. Additional troubleshooting was used by closing and reopening the Firefox app on the Apple devices to see if any changes were made to the bookmarks for the user profile after noticing the windows desktop app would initiate sync now frequently. This additional troubleshooting step yielded no positive results and the solution would be to manually click sync now on the source change device.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2001)

**Comment by SimonBasca** [comment]
Similar issue  #8663
